jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebInfConfigurationTest.java

```diff
@@ -110,4 +139,114 @@ public void testFindAndFilterContainerPathsTarget8()
         assertEquals(1, containerResources.size());
         assertThat(containerResources.get(0).toString(), containsString("jetty-util"));
     }
+
+    public static Stream<Arguments> baseResourceNames()
+    {
+        return Stream.of(
+            Arguments.of("test.war", "test.war"),
+            Arguments.of("a/b/c/test.war", "test.war"),
+            Arguments.of("bar%2Fbaz/test.war", "test.war"),
+            Arguments.of("fizz buzz/test.war", "test.war"),
+            Arguments.of("another one/bites the dust/", "bites the dust"),
+            Arguments.of("another+one/bites+the+dust/", "bites+the+dust"),
+            Arguments.of("another%20one/bites%20the%20dust/", "bites%20the%20dust"),
+            Arguments.of("spanish/número.war", "número.war"),
+            Arguments.of("spanish/n\uC3BAmero.war", "n\uC3BAmero.war"),
+            Arguments.of("spanish/n%C3%BAmero.war", "n%C3%BAmero.war"),
+            Arguments.of("a/b!/", "b!"),
+            Arguments.of("a/b!/c/", "c"),
+            Arguments.of("a/b!/c/d/", "d"),
+            Arguments.of("a/b%21/", "b%21")
+        );
```

Comment:
If the base resource dir is just something like "/a", or "/"?

Reply:
A `Resource` of just `/` or `/a` would be a `PathResource`.
That kind of `PathResource` would be valid on Linux and OSX, but not quite the same on Windows (as it will resolve to things like `C:\` or `C:\a`)

I'll add some test cases for those too.